amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/AwsCredentialsProviderPropertyValueDecoder.java

```diff
@@ -90,27 +94,32 @@ private static List<AWSCredentialsProvider> getValidCredentialsProviders(List<St
             }
             log.info("Attempting to construct {}", clazz);
 
-            AWSCredentialsProvider provider = null;
+            AwsCredentialsProvider provider = null;
             if (nameAndArgs.length > 1) {
                 final String[] varargs = Arrays.copyOfRange(nameAndArgs, 1, nameAndArgs.length);
 
                 // attempt to invoke an explicit N-arg constructor of FooClass(String, String, ...)
-                provider = constructProvider(providerName, () -> {
-                    Class<?>[] argTypes = new Class<?>[nameAndArgs.length - 1];
-                    Arrays.fill(argTypes, String.class);
-                    return clazz.getConstructor(argTypes).newInstance(varargs);
-                });
+                provider = constructProvider(
+                        providerName,
+                        () -> {
+                            Class<?>[] argTypes = new Class<?>[nameAndArgs.length - 1];
+                            Arrays.fill(argTypes, String.class);
+                            return clazz.getConstructor(argTypes).newInstance(varargs);
+                        },
+                        clazz);
```

Comment:
The new V2 providers are constructed using builders correct? Are they able use the same constructing logic as v1?

Reply:
New providers are constructed with builders / a `create()` call to build a generic instance, old providers were created with simple constructors.
<img width="1161" alt="Screenshot 2024-08-27 at 9 17 00 AM" src="https://github.com/user-attachments/assets/8b8c66de-478f-4438-a18c-2d4e0cd030e3">
<img width="1117" alt="Screenshot 2024-08-27 at 9 17 07 AM" src="https://github.com/user-attachments/assets/dfd45b75-20db-45e5-8aa1-6d8489eee152">

To build the old providers generically, we used `clazz::newInstance`. To build the new providers generically, my only idea was to do something like
```
Method createMethod = clazz.getDeclaredMethod("create");
Object provider = createMethod.invoke(null);
```

This makes things a bit complicated, as we are trying to generically build these classes in 2 different ways in one function. 

If you have a better idea to make this generic, I'm happy to hear it. The only other idea would be to implement a mapping like you mentioned [here](https://github.com/ethkatnic/amazon-kinesis-client/pull/1#discussion_r1732902876), and then call a constructOldProvider class if its a v1 provider and a constructNewProvider class if its a v2 provider and handle these separately.

Reply:
Simplified this a bit. Now [added an additional attempt to construct create()](https://github.com/ethkatnic/amazon-kinesis-client/pull/1/files#diff-6d02c70029ad7f983366ad4892c73feb18612ac4ee9ce1afbfbed9cb639d82baR120) without touching constructProvider().